src/com/sap/piper/GitUtils.groovy

```diff
@@ -5,60 +5,29 @@ boolean insideWorkTree() {
 }
 
 boolean isMergeCommit(){
-    def cmd = 'git rev-parse --verify HEAD^2'
-    return sh(returnStatus: true, script: cmd) == 0
-}
-
-String getGitMergeCommitId(String gitChangeId){
     if(!scm){
-        throw new Exception('scm content not found')
+        throw new Exception('scm context not found')
```

Comment:
(the following applies to all places where we through an exception in this PR)
I wonder:
* are we aware of specific scenarios where these error conditions would be realized?
* if the caller sees and catches the exception, is there any hope to recover from the problem?

Reply:
As per my understanding -
1. scm context should ideally exist. Users use "checkout scm" and then setupCommonPipelineEnv is called which is part of piper initialization. I have added these checks where there can be unknown issues.
2. pullrequest content depends on [plugin-content-pullrequest](https://plugins.jenkins.io/pipeline-github/#plugin-content-pullrequest) plugin. I could also see some references in our project with this plugin.
3. I can improve the logs for pullrequest content saying that the plugin should exist and should meet these requirements https://plugins.jenkins.io/pipeline-github/#plugin-content-prerequisites.